apis/oam/v1alpha2/register.go

```diff
@@ -0,0 +1,140 @@
+	"sigs.k8s.io/controller-runtime/pkg/scheme"
+)
+
+// Package type metadata.
+const (
+	Group   = "core.oam.dev"
+	Version = "v1alpha2"
+)
+
+var (
+	// SchemeGroupVersion is group version used to register these objects
+	SchemeGroupVersion = schema.GroupVersion{Group: Group, Version: Version}
+
+	// SchemeBuilder is used to add go types to the GroupVersionKind scheme
+	SchemeBuilder = &scheme.Builder{GroupVersion: SchemeGroupVersion}
+)
+
+// WorkloadDefinition type metadata.
+var (
+	WorkloadDefinitionKind             = reflect.TypeOf(WorkloadDefinition{}).Name()
+	WorkloadDefinitionGroupKind        = schema.GroupKind{Group: Group, Kind: WorkloadDefinitionKind}.String()
+	WorkloadDefinitionKindAPIVersion   = WorkloadDefinitionKind + "." + SchemeGroupVersion.String()
+	WorkloadDefinitionGroupVersionKind = SchemeGroupVersion.WithKind(WorkloadDefinitionKind)
+)
+
+// TraitDefinition type metadata.
+var (
+	TraitDefinitionKind             = reflect.TypeOf(TraitDefinition{}).Name()
+	TraitDefinitionGroupKind        = schema.GroupKind{Group: Group, Kind: TraitDefinitionKind}.String()
+	TraitDefinitionKindAPIVersion   = TraitDefinitionKind + "." + SchemeGroupVersion.String()
+	TraitDefinitionGroupVersionKind = SchemeGroupVersion.WithKind(TraitDefinitionKind)
+)
+
+// ScopeDefinition type metadata.
+var (
+	ScopeDefinitionKind             = reflect.TypeOf(ScopeDefinition{}).Name()
+	ScopeDefinitionGroupKind        = schema.GroupKind{Group: Group, Kind: ScopeDefinitionKind}.String()
+	ScopeDefinitionKindAPIVersion   = ScopeDefinitionKind + "." + SchemeGroupVersion.String()
+	ScopeDefinitionGroupVersionKind = SchemeGroupVersion.WithKind(ScopeDefinitionKind)
+)
+
+// Component type metadata.
+var (
+	ComponentKind             = reflect.TypeOf(Component{}).Name()
+	ComponentGroupKind        = schema.GroupKind{Group: Group, Kind: ComponentKind}.String()
+	ComponentKindAPIVersion   = ComponentKind + "." + SchemeGroupVersion.String()
+	ComponentGroupVersionKind = SchemeGroupVersion.WithKind(ComponentKind)
+)
+
+// ApplicationConfiguration type metadata.
+var (
+	ApplicationConfigurationKind             = reflect.TypeOf(ApplicationConfiguration{}).Name()
+	ApplicationConfigurationGroupKind        = schema.GroupKind{Group: Group, Kind: ApplicationConfigurationKind}.String()
+	ApplicationConfigurationKindAPIVersion   = ApplicationConfigurationKind + "." + SchemeGroupVersion.String()
+	ApplicationConfigurationGroupVersionKind = SchemeGroupVersion.WithKind(ApplicationConfigurationKind)
+)
+
+// Server type metadata.
+var (
+	ServerKind             = reflect.TypeOf(Server{}).Name()
+	ServerGroupKind        = schema.GroupKind{Group: Group, Kind: ServerKind}.String()
+	ServerKindAPIVersion   = ServerKind + "." + SchemeGroupVersion.String()
+	ServerGroupVersionKind = SchemeGroupVersion.WithKind(ServerKind)
+)
+
+// SingletonServer type metadata.
+var (
+	SingletonServerKind             = reflect.TypeOf(SingletonServer{}).Name()
```

Comment:
We are in the process of shrinking core types. We only need _Server_ as a container workload and _singleton_, _daemonset_, etc. are just developer traits (we called policies) of the workload.

Proposal: https://docs.google.com/document/d/1d3kVBFyf7yf-YzmVqkL1hr_qcIVXf6BUJLWKu9NJwvE/edit?usp=sharing

Reply:
My suggestion is to just define _Server_ for the comment.

Reply:
I believe we agreed on `ContainerisedWorkload` for the name; this is done.